src/backend/distributed/commands/alter_table.c

```diff
@@ -1057,8 +1057,11 @@ GetViewCreationCommandsOfTable(Oid relationId)
 		char *viewName = get_rel_name(viewOid);
 		char *schemaName = get_namespace_name(get_rel_namespace(viewOid));
 		char *qualifiedViewName = quote_qualified_identifier(schemaName, viewName);
+		bool isMatView = get_rel_relkind(viewOid) == RELKIND_MATVIEW;
+
 		appendStringInfo(query,
-						 "CREATE VIEW %s AS %s",
+						 "CREATE %s VIEW %s AS %s",
+						 isMatView ? "MATERIALIZED" : "",
```

Comment:
Let's also check if what we create is a materialized view in tests too.

Reply:
see https://github.com/citusdata/citus/pull/4584#discussion_r565216132